### ISSUE TITLE
Make `@CartesianProductTest` work with range sources

### ISSUE THREAD

**Issue by nipafx** [junit-pioneer/junit-pioneer#354]
It seems to me that `@CartesianProductTest` may work pretty well with the range sources - e.,g.:

```java
@CartesianProductTest(name = "word: {0} / number: {1}")
@CartesianValueSource(strings = { "foo", "bar"})
@IntRangeSource(from = 0, to = 10, step = 2)
void combination(String word, int number) { }
```

What do you think, @Michael1993? Let me know of you think this is worth exploring and whether you have time for it.

**Comment by Michael1993** [comment]
I looked at range sources when I was trying to work out how to reconcile `@CartesianProductTest` with `@ParameterizedTest` and had the same problem with it as with every other annotation that's annotated with `@ArgumentsSource`: they are not repeatable.

I made the decision that `@ParameterizedTest` enforcing a single `@ArgumentsSource` and `@CartesianProductTest` always needing multiple sources was too big of a design philosophy clash. Trying to make them work together would just lead to confusion.

However this is ultimately a one-way thing - `@CartesianProductTest` can not work with things designed for `@ParameterizedTest`. I think it would work the other way around - we could design repeatable `@ArgumentsSource`s for `@CartesianProductTest` that also happen to work with `@ParameterizedTest`.

TL;DR (the most important bit):
I will look into making range sources work with `@CartesianProductTest` while keeping it compatible with `@ParameterizedTest` and get back to you.

**Comment by nipafx** [comment]
But there's nothing stopping us from making the argument sources repeatable, right? What we would loose is the compiler support for the parameterized case, but if we would document that and generate a good error message when we encounter such a case, that would be ok I think.

**Comment by Michael1993** [comment]
Ha-ha-ha. `AnnotationSupport.findRepatableAnnotations` from JUnit 5 erases annotations that have the same values so I can't use it to find range sources. Very funny. :sob:
